src/components/single/SingleArticlePage.jsx

```diff
@@ -43,21 +46,55 @@ export const SingleArticlePage = () => {
                 username: newComment.author,
                 body: newComment.body,
             })
+            .then(posted => {
+                setCreatedCommentId(posted.data.comment.comment_id)
+            })
             .catch(err => {
-                setComments((currComments) => [currComments.shift(), ...currComments])
+                setComments((currComments) => {
+                    return [...currComments].filter(comment => comment.comment_id !== undefined)
+                })
                 console.log(err)
-                toast.error("Your comment hasn't been published!")
+                toast.error("Your comment hasn't been published!", {
+                    toastId: "error"        
+                })
             })
         }
     }, [newComment])
 
+    useEffect(() => {
+        setComments((currComments) => {
+            return [...currComments].filter(comment => {
+                if (comment.comment_id == undefined) {
+                    comment.comment_id = createdCommentId
+                } else {
+                    return comment
+                }
+            })
+        })
+    }, [createdCommentId])
+
+    useEffect(() => {
+        if (deletedComment.comment_id) {
+            setTimeout(() => setComments((currComments) => {
+                return [...currComments].filter(comment => comment.comment_id !== deletedComment.comment_id)
+            }), 500)
+            deleteComment(deletedComment.comment_id)
+            .catch(err => {
+                console.log(err)
+                setComments((currComments) => [deletedComment, ...currComments])
+                toast.error("Your comment hasn't been deleted!", {
+                    toastId: "error"        
+                })
+            })
+        } 
+    }, [deletedComment])
```

Comment:
This useEffect isn't necessary. All the functionality you do in here can be done directly instead of doing `setDeletedComment(comment)` to trigger it. 

You can remove that state of `deletedComment` and this useEffect and do this directly.